### PR TITLE
optimize matrix sync policy

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

**Review by AviAvni — APPROVED** [review]

File: src/execution_plan/ops/shared/create_functions.c [hunk 1]
```diff
@@ -21,78 +21,106 @@ static inline void _AddProperties(ResultSetStatistics *stats, GraphEntity *ge,
...
 		const char *relation = edge_ctx->relation;
-		if(relation) {
-			if(GraphContext_GetSchema(gc, relation, SCHEMA_EDGE) == NULL) {
-				GraphContext_AddSchema(gc, relation, SCHEMA_EDGE);
-			}
-		}
+		Schema *s = GraphContext_GetSchema(gc, relation, SCHEMA_EDGE);
+		if(s == NULL) s = GraphContext_AddSchema(gc, relation, SCHEMA_EDGE);
+
+		Graph_GetRelationMatrix(g, s->id, false);
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/shared/create_functions.c:109]: Add explanatory comment about syncing matrix

File: src/execution_plan/ops/shared/create_functions.c [hunk 2]
```diff
@@ -21,78 +21,106 @@ static inline void _AddProperties(ResultSetStatistics *stats, GraphEntity *ge,
...
 	}
 
-	uint edge_count = array_len(pending->created_edges);
-	Graph_AllocateEdges(g, edge_count);
+	Graph_GetAdjacencyMatrix(g, false);
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/shared/create_functions.c:112]: Add explanatory comment about syncing matrix

File: src/execution_plan/ops/shared/create_functions.c [hunk 3]
```diff
@@ -142,19 +174,41 @@ void CommitNewEntities(OpBase *op, PendingCreations *pending) {
...
+		Graph_AllocateNodes(g, node_count);
+
+		// set graph matrix sync policy to resize
+		// no need to perform sync
+		Graph_SetMatrixPolicy(g, SYNC_POLICY_RESIZE);
+		_CommitNodesBlueprint(pending);
+
+		// set graph matrix sync policy to NOP
+		// no need to perform sync/resize
+		Graph_SetMatrixPolicy(g, SYNC_POLICY_NOP);
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/shared/create_functions.c:187]: This is very nice!

**Review by jeffreylovitz — DISMISSED** [review]
This is a really nice change, avoiding unnecessary syncs in the creation routines is great!

**Pull request merged**